src/graphdb.js

```diff
@@ -52,74 +52,157 @@ function toPascalCase (str) {
     return r.trim();
 }
 
+// Changes every instance of invalid characters in the given label with the following abbreviations
+function replaceCleanseLabel(label) {
+    return label
+        .replaceAll("!", "_ex_")
+        .replaceAll("$", "_dol_")
+        .replaceAll("&", "_amp_")
+        .replaceAll("(", "_op_")
+        .replaceAll(")", "_cp_")
+        .replaceAll(".", "_dot_")
+        .replaceAll(":", "_cn_")
+        .replaceAll("=", "_eq_")
+        .replaceAll("@", "_at_")
+        .replaceAll("[", "_os_")
+        .replaceAll("]", "_cs_")
+        .replaceAll("{", "_oc_")
+        .replaceAll("|", "_vb_")
+        .replaceAll("}", "_cc_")
+        .replaceAll("-", "_hy_");
+}
 
 function graphDBInferenceSchema (graphbSchema, addMutations) {
     let r = '';
+    let invalidNode = false;
     const gdbs = JSON.parse(graphbSchema);
 
     checkForDuplicateNames(gdbs);
 
     gdbs.nodeStructures.forEach(node => {
         // node type
-        let nodeCase = node.label;
-        if (changeCase) {
-            nodeCase = toPascalCase(node.label); 
+        let nodeCase = replaceCleanseLabel(node.label);
```

Comment:
We can simplify the if statement by first changing to pascal case first and then checking if there was a difference compared to node label. That way an alias will be added regardless if there was special character handling and/or case change to pascal case:

```
let nodeCase = replaceCleanseLabel(node.label);
        if (changeCase) {
            nodeCase = toPascalCase(node.label);
        }
        if (node.label !== nodeCase) {
            r += `type ${nodeCase} @alias(property:"${node.label}") {\n`;
        } else {
            r += `type ${nodeCase} {\n`;
        }
```

Reply:
Sorry the above suggestion has a problem, trying to fix it!